apitoolkit_pyramid/__init__.py

```diff
@@ -82,8 +100,14 @@ def process_exception(self, request, exception):
         pass
 
     def __call__(self, request: Request):
+        response = self.get_response(request)
         if self.debug:
             print("APIToolkit: making request")
+        if self.routes_whitelist:
+            # return early when route does not match any of the whitelist routes
+            if not any([request.path.startswith(route) for route in self.routes_whitelist]):
```

Comment:
This solution might encounter limitations when dealing with routes incorporating path parameters. For instance, if we include `/product/{category}` in the whitelist, the existing logic may overlook URLs like `/product/electronics`. 

To address this, we can refine our approach by updating the logic to consider the matched route pattern more comprehensively. We currently have:
```python
url_path = request.matched_route.pattern if request.matched_route is not None else request.path
```
We should consider checking against `url_path` instead of `request.path`. This adjustment should  resolve the issue.